dialog/fileicon.go

```diff
@@ -31,7 +31,7 @@ const ratioDown float64 = 0.45
 // NewFileIcon takes a filepath and creates an icon with an overlayed label using the detected mimetype and extension
 func NewFileIcon(path string) fyne.CanvasObject {
 
-	ext, mimeType, mimeSubType := mimeTypeGet(path)
+	ext, mimeType, mimeSubType := mimeTypeGet(storage.NewURI(path))
```

Comment:
I hadn't noticed that this API was public last time - given that we use URI everywhere else and we need it internally, this should probably have taken a URI parameter not a path string.

Reply:
Alright I fixed that. And made NewFileIcon take a URI as well.